pkg/kubelet/kubelet_node_status.go

```diff
@@ -370,13 +369,26 @@ func (kl *Kubelet) tryUpdateNodeStatus(tryNumber int) error {
 	// apiserver cache (the data might be slightly delayed but it doesn't
 	// seem to cause more conflict - the delays are pretty small).
 	// If it result in a conflict, all retries are served directly from etcd.
-	opts := metav1.GetOptions{}
+	var node *v1.Node
+	var err error
 	if tryNumber == 0 {
-		util.FromApiserverCache(&opts)
+		node, err = kl.nodeInfo.GetNodeInfo(string(kl.nodeName))
```

Comment:
rather than set `node` and then clear, prefer setting a local var and then assigning to `node` when heartbeat matches:

```
cachedNode, err := kl.nodeInfo.GetNodeInfo(string(kl.nodeName))
switch {
case err != nil:
  glog.Warningf("error getting node from cache %q: %v", kl.nodeName, err)
case !kl.getNodeReadyLastHeartbeatTime(cachedNode).Equal(kl.lastHeartbeatTime):
  glog.Infof("cached node heartbeat time does not match last submitted time")
default:
  node = cachedNode
}
```